pallets/ocex/src/lib.rs

```diff
@@ -1233,6 +1232,70 @@ pub mod pallet {
 				false
 			}
 		}
+
+		pub fn get_ob_recover_state() -> Result<
+			(
+				BTreeMap<AccountId, Vec<AccountId>>,
+				BTreeMap<AccountAsset, Decimal>,
+				u64,
+				u32,
+				u64,
+				u64,
+			),
+			DispatchError,
+		> {
+			let account_id =
+				<Accounts<T>>::iter().fold(vec![], |mut ids_accum, (acc, acc_info)| {
+					ids_accum.push((acc.clone(), acc_info.proxies));
+					ids_accum
+				});
+
+			let mut balances: BTreeMap<AccountAsset, Decimal> = BTreeMap::new();
+			let mut account_ids: BTreeMap<AccountId, Vec<AccountId>> = BTreeMap::new();
+			// all offchain balances for main accounts
+			for account in account_id {
+				let main = Self::transform_account(account.0)?;
+				let b = Self::get_offchain_balance(&main)?;
+				for (asset, balance) in b.into_iter() {
+					balances.insert(AccountAsset { main: main.clone(), asset }, balance);
+				}
+				let proxies = account.1.into_iter().try_fold(vec![], |mut accum, proxy| {
+					accum.push(Self::transform_account(proxy)?);
+					Ok::<Vec<AccountId>, DispatchError>(accum)
+				})?;
+				account_ids.insert(main, proxies);
+			}
+
+			let state_info = Self::get_state_info();
+			let last_processed_block_number = state_info.last_block;
+			let worker_nonce = state_info.worker_nonce;
+			let snapshot_id = state_info.snapshot_id;
+			let state_change_id = state_info.stid;
+
+			Ok((
+				account_ids,
+				balances,
+				snapshot_id,
+				last_processed_block_number,
+				state_change_id,
+				worker_nonce,
+			))
+		}
+
+		pub fn get_balance(from: T::AccountId, of: AssetId) -> Result<Decimal, DispatchError> {
```

Comment:
Please add some documentation for the public api.

Reply:
done